pkg/core/mempool/mempool.go

```diff
@@ -124,53 +137,66 @@ func (m *Mempool) Run() {
 		for {
 			select {
 			case r := <-m.sendTxChan:
-				m.onSendTx(r)
+				handleRequest(r, m.onSendMempoolTx, "SendTx")
 			case r := <-m.getMempoolTxsChan:
-				m.onGetMempoolTxs(r)
+				handleRequest(r, m.onGetMempoolTxs, "GetMempoolTxs")
 			// Mempool input channels
 			case b := <-m.acceptedBlockChan:
 				m.onAcceptedBlock(b)
 			case tx := <-m.pending:
-				txid, err := m.onPendingTx(tx)
-				if err != nil {
-					log := logEntry("tx", toHex(txid[:]))
-					log.Errorf("%v", err)
-				}
+				_, _ = m.onPendingTx(tx)
 			case <-time.After(20 * time.Second):
 				m.onIdle()
 			// Mempool terminating
 			case <-m.quitChan:
 				return
-
 			}
 		}
 	}()
 }
 
-// onPendingTx ensures all transaction rules are satisfied before adding the tx
-// into the verified pool
+// onPendingTx handles a submitted tx from any source (rpcBus or eventBus)
 func (m *Mempool) onPendingTx(t TxDesc) ([]byte, error) {
```

Comment:
`onPendingTx` should not return values since errors are handled directly in the method body and the `txid` is never checked anywhere by the client of the function

Reply:
`onPendingTx` return values are needed by `onSendMempoolTx` to enable rpcBus call to retrieve txid or error on submitting a tx to the pool.